source/common/network/address_impl.cc

```diff
@@ -377,7 +380,14 @@ Api::SysCallIntResult PipeInstance::bind(int fd) const {
     unlink(address_.sun_path);
   }
   auto& os_syscalls = Api::OsSysCallsSingleton::get();
-  return os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  auto bind_result = os_syscalls.bind(fd, sockAddr(), sockAddrLen());
+  if (mode != 0 and !abstract_namespace_ and bind_result.rc_ == 0) {
```

Comment:
probably doesn't matter here but use `&&` instead of and for consistency (MSVC doesn't support them)